### PR TITLE
dubbo zookeeper registry too slow

### PR DESCRIPTION
**Pull request by qixiaobo**
ref to #4828
I just add concurrentHashSet to avoid checking persistent node exist

### CODE REVIEW

File: dubbo-remoting/dubbo-remoting-zookeeper/src/main/java/org/apache/dubbo/remoting/zookeeper/support/AbstractZookeeperClient.java [hunk 1]
```diff
@@ -45,6 +46,8 @@
 
     private volatile boolean closed = false;
 
+    private final Set<String>  persistentExistNodePath = Sets.newConcurrentHashSet();
+
```
Comment (tswstarplanet) [inline dubbo-remoting/dubbo-remoting-zookeeper/src/main/java/org/apache/dubbo/remoting/zookeeper/support/AbstractZookeeperClient.java]: If a path is deleted, the path should be removed from the set.

Comment (qixiaobo) [reply dubbo-remoting/dubbo-remoting-zookeeper/src/main/java/org/apache/dubbo/remoting/zookeeper/support/AbstractZookeeperClient.java]: > If a path is deleted, the path should be removed from the set.
From me , I think we delete only the non persistent node. I will have a look with the code

**Comment by qixiaobo** [comment]
@tswstarplanet  You are right, I find the metaDataReport delete the persist node

File: dubbo-remoting/dubbo-remoting-zookeeper/src/main/java/org/apache/dubbo/remoting/zookeeper/support/AbstractZookeeperClient.java (hunk 1 above)
Comment (beiwei30) [reply dubbo-remoting/dubbo-remoting-zookeeper/src/main/java/org/apache/dubbo/remoting/zookeeper/support/AbstractZookeeperClient.java]: pls. avoid of using guava. It's introduced from

File: dubbo-remoting/dubbo-remoting-zookeeper/src/main/java/org/apache/dubbo/remoting/zookeeper/support/AbstractZookeeperClient.java [hunk 2]
```diff
@@ -16,6 +16,7 @@
  */
 package org.apache.dubbo.remoting.zookeeper.support;
 
+import com.google.common.collect.Sets;
```
Comment (beiwei30) [inline dubbo-remoting/dubbo-remoting-zookeeper/src/main/java/org/apache/dubbo/remoting/zookeeper/support/AbstractZookeeperClient.java]: pls. avoid of using guava directly.

Comment (qixiaobo) [reply dubbo-remoting/dubbo-remoting-zookeeper/src/main/java/org/apache/dubbo/remoting/zookeeper/support/AbstractZookeeperClient.java]: OK, I will modify the code

**Pull request merged**